File: lintcode/easy/unique_characters/cpp/unique_characters.cpp
/*
@Copyright:LintCode
@Author:   lilsweetcaligula
@Problem:  http://www.lintcode.com/problem/unique-characters
@Language: C++
@Datetime: 17-02-15 18:27
*/

#include <algorithm>

class Solution {
public:
    /**
     * @param str: a string
     * @return: a boolean
     */
    bool isUnique(std::string str) 
    {
        std::sort(str.begin(), str.end());
        
        for (decltype(str.size()) i{ 1 }; i < str.size(); ++i) {
            if (str[i] == str[i - 1]) {
                return false;
            }
        }
        
        return true;
    }
};

File: lintcode/easy/unique_characters/py/unique_characters.py
# coding:utf-8
'''
@Copyright:LintCode
@Author:   lilsweetcaligula
@Problem:  http://www.lintcode.com/problem/unique-characters
@Language: Python
@Datetime: 17-02-15 14:58
'''

class Solution:
    # @param s: a string
    # @return: a boolean
    def isUnique(self, s):
        ordered = sorted(s)
        
        for i in range(1, len(ordered)):
            if ordered[i] == ordered[i - 1]:
                return False
                
        return True
    